numpyro/infer/util.py

```diff
@@ -108,14 +109,17 @@ def constrain_fn(model, transforms, model_args, model_kwargs, params, return_det
         sites from the model. Defaults to `False`.
     :return: `dict` of transformed params.
     """
-    params_constrained = transform_fn(transforms, params)
-    substituted_model = substitute(model, param_map=params_constrained)
+    def substitute_fn(site):
+        if site['name'] in params:
+            return biject_to(site['fn'].support)(params[site['name']])
```

Comment:
@neerajprad This is unconstrained reparam that I mentioned. This way, we don't need to store `inv_transforms`.